core/types/block.go

```diff
@@ -466,6 +479,21 @@ func (b *Block) WithBody(body Body) *Block {
 	return block
 }
 
+func (b *Block) WithWitness(witness *ExecutionWitness) *Block {
+
+	block := &Block{
+		header:       b.header,
+		transactions: slices.Clone(b.transactions),
```

Comment:
It's not necessary to copy the block's component fields here. You can just assign them directly.